### PR TITLE
Bigquery: Add Table Level Tags, fix dataset issue

### LINKED ISSUE #12738: [Bug] Dataset information (BigQuery)
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
Backend

**Describe the bug**
When ingesting the dataset information an error occur during it.

```
[2023-08-03, 17:56:57 UTC] {topology_runner.py:77} DEBUG - Processing stage: type_=<class 'metadata.generated.schema.entity.data.databaseSchema.DatabaseSchema'> processor='yield_database_schema' context='database_schema' ack_sink=True nullable=False must_return=False cache_all=False clear_cache=False overwrite=True consumer=['database_service', 'database']
[2023-08-03, 17:56:57 UTC] {metadata.py:256} DEBUG - Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/metadata/ingestion/source/database/bigquery/metadata.py", line 251, in get_schema_description
    query_result = [result.schema_description for result in query_resp.result()]
  File "/home/airflow/.local/lib/python3.9/site-packages/google/cloud/bigquery/job/query.py", line 1520, in result
    do_get_result()
  File "/home/airflow/.local/lib/python3.9/site-packages/google/api_core/retry.py", line 283, in retry_wrapped_func
    return retry_target(
  File "/home/airflow/.local/lib/python3.9/site-packages/google/api_core/retry.py", line 190, in retry_target
    return target()
  File "/home/airflow/.local/lib/python3.9/site-packages/google/cloud/bigquery/job/query.py", line 1510, in do_get_result
    super(QueryJob, self).result(retry=retry, timeout=timeout)
  File "/home/airflow/.local/lib/python3.9/site-packages/google/cloud/bigquery/job/base.py", line 922, in result
    return super(_AsyncJob, self).result(timeout=timeout, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/google/api_core/future/polling.py", line 137, in result
    raise self._exception
google.api_core.exceptions.BadRequest: 400 Syntax error: Expected end of input but got "-" at [3:23]


[2023-08-03, 17:56:57 UTC] {metadata.py:257} DEBUG - Failed to fetch dataset description for [dataset_test]: 400 Syntax error: Expected end of input but got "-" at [3:23]
```

**To Reproduce**

Screenshots or steps to reproduce

**Expected behavior**

**Version:**
 - OS: ubuntu
 - Python version:
 - OpenMetadata version: 1.1.0
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### LINKED ISSUE #12879: [feature-request] Collect bigquery tables tags
**Feature**
BigQuery does not ingest information such as labels from tables

**Describe the task**
Collect the labels from table during metadata ingestion. Its the same way used for datasets.

image

### LINKED ISSUE #12989: When setting multi project in bigquery database, metadata ingestion fails with an 404 error.
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
Ingestion Framework

**Describe the bug**

In bigquery database settings, I want to set Multiple Project ID to get schema and lineage from all related projects.

Referring to https://docs.open-metadata.org/v1.1.1/connectors/database/bigquery, I set two Multiple Project IDs.

1. main-project
2. sub-project

We created a service account in main project and enabled the bigquery role in sub-project by referring to the following document.
Referenced documentation : https://www.cloudquery.io/blog/creating-cross-project-service-accounts-in-gcp
Using a single service account, we see that we can access both projects.

I'm running metadata scraping, but I'm getting the following error.

The error that the main project schema and tables are not found in the subproject and , error that the subproject schema and tables are not found in the main project.

ERROR log example

NotFound: 404 GET https://bigquery.googleapis.com/bigquery/v2/projects/{sub-project}/datasets/{schema}/tables/{main-project-table}`?prettyPrint=false: Not found:

```
google.api_core.exceptions.NotFound: 404 GET https://bigquery.googleapis.com/bigquery/v2/projects/bigquery-project-external/datasets/dataset/tables/example-table?prettyPrint=false: Not found: Table bigquery-project-external:dataset.example-table
[2023-08-24, 09:15:34 UTC] {common_db_source.py:419} WARNING - Unexpected exception to yield table [example-table]: 404 GET https://bigquery.googleapis.com/bigquery/v2/projects/bigquery-project-external/datasets/dataset/tables/example-table?prettyPrint=false: Not found: Table bigquery-project-external:dataset.example-table
[2023-08-24, 09:15:34 UTC] {helpers.py:128} DEBUG - yield_table executed in 0.61s
[2023-08-24, 09:15:34 UTC] {topology_runner.py:77} DEBUG - Processing stage: type_=<class 'metadata.generated.schema.entity.data.table.Table'> processor='yield_table' context='table' ack_sink=True nullable=False must_return=False cache_all=False clear_cache=False overwrite=True consumer=['database_service', 'database', 'database_schema']
[2023-08-24, 09:15:34 UTC] {sql_column_handler.py:119} DEBUG - Cannot obtain unique constraints for table [example-table]: NotImplementedError
[2023-08-24, 09:15:35 UTC] {common_db_source.py:418} DEBUG - Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/metadata/ingestion/source/database/common_db_source.py", line 395, in yield_table
    is_partitioned, partition_details = self.get_table_partition_details(
  File "/home/airflow/.local/lib/python3.9/site-packages/metadata/ingestion/source/database/bigquery/metadata.py", line 394, in get_table_partition_details
    table = self.client.get_table(f"{database}.{schema_name}.{table_name}")
  File "/home/airflow/.local/lib/python3.9/site-packages/google/cloud/bigquery/client.py", line 1034, in get_table
    api_response = self._call_api(
  File "/home/airflow/.local/lib/python3.9/site-packages/google/cloud/bigquery/client.py", line 782, in _call_api
    return call()
  File "/home/airflow/.local/lib/python3.9/site-packages/google/api_core/retry.py", line 283, in retry_wrapped_func
    return retry_target(
  File "/home/airflow/.local/lib/python3.9/site-packages/google/api_core/retry.py", line 190, in retry_target
    return target()
  File "/home/airflow/.local/lib/python3.9/site-packages/google/cloud/_http/__init__.py", line 494, in api_request
    raise exceptions.from_http_response(response)
google.api_core.exceptions.NotFound: 404 GET https://bigquery.googleapis.com/bigquery/v2/projects/bigquery-project-external/datasets/dataset/tables/example-table?prettyPrint=false: Not found: Table bigquery-project-external:example-table
[2023-08-24, 09:15:35 UTC] {common_db_source.py:419} WARNING - Unexpected exception to yield table [example-table]: 404 GET https://bigquery.googleapis.com/bigquery/v2/projects/bigquery-project-external/datasets/dataset/tables/example-table?prettyPrint=false: Not found: Table bigquery-project-external:example-table
[2023-08-24, 09:15:35 UTC] {helpers.py:128} DEBUG - yield_table executed in 0.53s
[2023-08-24, 09:15:35 UTC] {topology_runner.py:77} DEBUG - Processing stage: type_=<class 'metadata.generated.schema.entity.data.table.Table'> processor='yield_table' context='table' ack_sink=True nullable=False must_return=False cache_all=False clear_cache=False overwrite=True consumer=['database_service', 'database', 'database_schema']
[2023-08-24, 09:15:35 UTC] {sql_column_handler.py:119} DEBUG - Cannot obtain unique constraints for table [dataset.table_example_table]: NotImplementedError
[2023-08-24, 09:15:35 UTC] {common_db_source.py:418} DEBUG - Traceback (most recent call last):
```

1. it works fine when only MAIN PROJECT is set.
2. If I set up database service separately for main project and sub project, it works normally.
3. If I set both main project and sub project in one database service, the error mentioned above occurs.

**To Reproduce**

set both main project and sub project in one database service, the error mentioned above occurs.

**Expected behavior**

**Version:**
 - OS: [e.g. iOS] `docker image 1.1.1`, on kubernetes ENV
 - Python version: `docker image 1.1.1` 
 - OpenMetadata version: [e.g. 0.8] `1.1.1`
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`] `docker image 1.1.1` 

**Additional context**
slack thread : https://openmetadata.slack.com/archives/C02B6955S4S/p1692602059966489

### PR DESCRIPTION
**Pull request by ayush-shah**
### Describe your changes:

Fixes #12879 
Fixes #12738 

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/database/bigquery/metadata.py [hunk 1]
```diff
@@ -327,12 +329,40 @@ def yield_database_schema(
                 )
         yield Either(right=database_schema_request_obj)
 
+    def get_table_obj(self, table_name: str):
+        schema_name = self.context.database_schema.name.__root__
+        database = self.context.database.name.__root__
+        return self.client.get_table(f"{database}.{schema_name}.{table_name}")
```
Comment (OnkarVO7) [inline ingestion/src/metadata/ingestion/source/database/bigquery/metadata.py:335]: We can use the fqn builder here

**Review by OnkarVO7 — CHANGES_REQUESTED** [review]

File: ingestion/src/metadata/ingestion/source/database/bigquery/metadata.py (hunk 1 above)
Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/source/database/bigquery/metadata.py:337]: table fqn won’t be needed there because the client needs project.schema.table and not dbservice.project.schema.table we won’t be able to use fqn.build

Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/source/database/bigquery/metadata.py]: used fqn._build as suggested, thanks

**Review by OnkarVO7 — APPROVED** [review]

File: ingestion/src/metadata/profiler/interface/sqlalchemy/bigquery/profiler_interface.py [hunk 2]
```diff
@@ -43,12 +76,18 @@ def _get_struct_columns(self, columns: dict, parent: str):
 
     def _get_sampler(self, **kwargs):
         """get sampler object"""
-        session = kwargs.get("session")
```
Comment (TeddyCr) [inline ingestion/src/metadata/profiler/interface/sqlalchemy/bigquery/profiler_interface.py:46]: why did you remove this?

File: ingestion/src/metadata/profiler/interface/sqlalchemy/profiler_interface.py [hunk 3]
```diff
@@ -96,6 +96,9 @@ def __init__(
         )
 
         self._table = self._convert_table_to_orm_object(sqa_metadata)
+        self.set_session()
+
+    def set_session(self):
         self.session_factory = self._session_factory()
         self.session = self.session_factory()
         self.set_session_tag(self.session)
```
Comment (TeddyCr) [inline ingestion/src/metadata/profiler/interface/sqlalchemy/profiler_interface.py:104]: unsure if we should do that. I know we do it in quite some places but it is generally not a good practice to instantiate object attributes outside of the `__init__`

File: ingestion/src/metadata/profiler/interface/sqlalchemy/bigquery/profiler_interface.py [hunk 4]
```diff
@@ -24,6 +27,36 @@
...
+            profile_sample_config,
+            source_config,
+            sample_query,
+            table_partition_config,
+            thread_count,
+            timeout_seconds,
+            sqa_metadata,
+            **kwargs,
+        )
+        self.specific_client = {}
```
Comment (TeddyCr) [inline ingestion/src/metadata/profiler/interface/sqlalchemy/bigquery/profiler_interface.py:58]: why do we do that with `self.specific_client = {}`?

File: ingestion/src/metadata/profiler/interface/sqlalchemy/profiler_interface.py (hunk 3 above)
Comment (ayush-shah) [reply ingestion/src/metadata/profiler/interface/sqlalchemy/profiler_interface.py:104]: the method instantiates the object attributes which is called from __init__ itself and instantiates before being used, and also can be used again to instantiate object attributes with new connection in different methods..

File: ingestion/src/metadata/profiler/interface/sqlalchemy/bigquery/profiler_interface.py (hunk 2 above)
Comment (ayush-shah) [reply ingestion/src/metadata/profiler/interface/sqlalchemy/bigquery/profiler_interface.py:46]: actually, just found out a method to do overriding, we won't need to remove this. will add in new commit

File: ingestion/src/metadata/profiler/interface/sqlalchemy/profiler_interface.py (hunk 3 above)
Comment (ayush-shah) [reply ingestion/src/metadata/profiler/interface/sqlalchemy/profiler_interface.py:104]: reverted this as it's not needed, thanks

**Review by TeddyCr — APPROVED** [review]
